src/signer/mod.rs

```diff
@@ -80,23 +254,38 @@ pub trait SignerTrait {
         Ok(SignedTransaction::new(signature, unsigned_transaction))
     }
 
+    /// Creates an unsigned transaction and returns it along with the secret key.
+    /// This is a `helper` method that should be implemented by the signer or fail with SignerError.
+    /// As long as this method works, the default implementation of the [sign_meta](`SignerTrait::sign_meta`) and [sign](`SignerTrait::sign`) methods should work.
+    ///
+    /// If you can't provide a SecretKey for some reason (E.g. Ledger),
+    /// you can fail with SignerError and override `sign_meta` and `sign` methods.
     fn tx_and_secret(
```

Comment:
Maybe not following, but I would consider using `signer_id`, `receiveer_id` and `actions` parameters instead of `PrepopulateTransaction` in the future

Reply:
Why ?

Reply:
PrepopulateTransaction is not a common term and is a bit confusing. But generally not that bad